frontend/docs/about-slow-types.md

```diff
@@ -367,6 +367,37 @@ adding the following to your `deno.json` configuration file:
 }
 ```
 
-Note that because slow types can not be individually ignored, one can not use an
-ignore comment like `// deno-lint-ignore no-slow-types` to ignore slow type
-diagnostics.
+Note that because slow type diagnostics can not be individually ignored, one can
+not use an ignore comment like `// deno-lint-ignore no-slow-types` to ignore
+slow type diagnostics.
+
+## Interactions with TypeScript `isolatedDeclarations`
+
+Since TypeScript 5.5, TypeScript has introduced a compiler option called
+`isolatedDeclarations`. When enabled, this option disallows writing types that
+would require type inference to emit a declaration file. This is very similar to
+the "no slow types" policy of JSR.
+
+For example, just like JSR, TypeScript with `isolatedDeclarations` enabled would
+not allow funtion declarations without an explicit return type:
+
+```ts
+// This is not allowed with `isolatedDeclarations`.
+export function foo() {
+  return Math.random().toString();
+}
+```
+
+However, there are some differences between the two:
+
+- Isolated declarations require that all symbols that are exported from a module
+  follow the "no inference" rules. JSR only requires that symbos that are
+  actually part of the public API of a package follow the "no inference" rules.
+- Isolated declarations is sometimes more strict than JSR. For example, isolated
```

Comment:
Nit: Should we show an example of what that looks like? Not sure if JS devs are familiar with the term `function expression body`.